simapp/simd/cmd/testnet.go

```diff
@@ -398,3 +469,43 @@ func writeFile(name string, dir string, contents []byte) error {
 
 	return nil
 }
+
+// StartTestnet starts an in-process testnet
+func StartTestnet(cmd *cobra.Command, testnetsDir string, chainID string, minGasPrices string, algo string,
+	numValidators int, enableLogging bool, rpcAddress, apiAddress, grpcAddress string, printMnemonic bool) error {
+	networkConfig := network.DefaultConfig()
+
+	// Default networkConfig.ChainID is random, and we should only override it if chainID provided
+	// is non-empty
+	if chainID != "" {
+		networkConfig.ChainID = chainID
+	}
+	networkConfig.SigningAlgo = algo
+	networkConfig.MinGasPrices = minGasPrices
+	networkConfig.NumValidators = numValidators
+	networkConfig.EnableTMLogging = enableLogging
+	networkConfig.RPCAddress = rpcAddress
+	networkConfig.APIAddress = apiAddress
+	networkConfig.GRPCAddress = grpcAddress
+	networkConfig.PrintMnemonic = printMnemonic
+	networkLogger := network.NewCLILogger(cmd)
+
+	baseDir := fmt.Sprintf("%s/%s", testnetsDir, networkConfig.ChainID)
```

Comment:
The current behavior is:
- the `start` command runs in a `.testnets/${chainid}` subfolder
- whereas the `init-files` command creates in the `.testnets` root folder, right?

Reply:
Yes, this is correct. It might not be necessary to use the chain-id subfolder for the `start` command since we clean it up anyways but it would also be nice to use the chain-id subfolder for both (but this complicated the `make localnet-start` command in a way that was not ideal).